hypothesis-python/tests/array_api/test_indices.py

```diff
@@ -73,23 +89,39 @@ def test_efficiently_generate_indexers(xp, xps):
     find_any(xps.indices((3, 3, 3, 3, 3)))
 
 
-@given(allow_ellipsis=st.booleans(), data=st.data())
-def test_generate_valid_indices(xp, xps, allow_ellipsis, data):
+@given(allow_newaxis=st.booleans(), allow_ellipsis=st.booleans(), data=st.data())
+def test_generate_valid_indices(xp, xps, allow_newaxis, allow_ellipsis, data):
     """Strategy generates valid indices."""
+    try:
+        from numpy import array_api as nxp
+
+        if xp == nxp:
+            pytest.xfail(
+                "numpy.array_api arrays do not currently support newaxis indexing"
+            )
+    except ImportError:
+        pass
```

Comment:
```suggestion
    try:
        import numpy.array_api
    except ImportError:
        pass
    else:
        if xp is numpy.array_api:
            assert <doesn't support newaxis indexing yet>, "We want tests to fail when we can delete this whole special case"
            allow_newaxis = False
```

Reply:
Ah I see, pushed a change that does this. Instead of using say `pytest.raises(IndexError)` on a newaxis operation, I thought to use another try-except-else block to raise an exception w/ an immediate call to action.

Reply:
Looks great!